hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsCompressor.java

```diff
@@ -329,13 +339,32 @@ private void reset(int estimatedBytesDictionary, int estimatedBytesMetrics) {
         lastDescriptor = null;
     }
 
-    private byte[] getRenderedBlob() {
+    /**
+     * Frees resources associated with this compressor. Needed to avoid keeping
+     * memory allocated by {@link Deflater} for a long time, until finalization.
+     */
+    public void close() {
         try {
-            writeDictionary();
             dictionaryDos.close();
```

Comment:
Although dictionaryDos and metricDos should be non null in most cases, IMHO adding null check here will make us sure the cleanup won't fail

Reply:
I added `@Nonnull` instead. These fields never should be null. However, IntelliJ is complaining because they are not initialised directly in the constructor.